core/include/scipp/core/string.h

```diff
@@ -92,6 +92,9 @@ template <class T> std::string element_to_string(const T &item) {
   else if constexpr (std::is_same_v<T, Eigen::Vector3d>)
     return {"(" + to_string(item[0]) + ", " + to_string(item[1]) + ", " +
             to_string(item[2]) + "), "};
+  else if constexpr (std::is_same_v<T, Eigen::Quaterniond>)
+    return {"(" + to_string(item.x()) + ", " + to_string(item.y()) + ", " +
```

Comment:
Discussion point - We provide quaternions here for the purposes of supporting rotations. For human readability an axis + axis angle representation, or even a rotation matrix might be easier to quickly understand.

Also, this returns the un-labeled elements of the quaternion. in the order x, y, z, w. This fits with the above `Vector3d` representation. In both cases, should we explicitly state what each element refers to in the return value? I suspect that would be too noisy. 

I think what you've done here is probably right.

Reply:
I also think having `x=..., y=...` is too much. For Vector it's actually transparent. For Quaternions, I agree that it's more ambiguous, as it's not clear whether the convention is `x, y, z, w` or `w, x, y, z`. Maybe this can just be made clear in the documentation.

Btw, should I add some documentation on the quaternions?